eng/testing/linker/SupportFiles/Directory.Build.targets

```diff
@@ -0,0 +1,25 @@
+<Project>
+  
+  <PropertyGroup>
+    <!-- Used to silence the warning caused by the workaround for https://github.com/dotnet/runtime/issues/81382 -->
+    <SuppressGenerateILCompilerExplicitPackageReferenceWarning>true</SuppressGenerateILCompilerExplicitPackageReferenceWarning>
+  </PropertyGroup>
+
+  <!-- needed to reference a specific version of NetCoreApp. Workaround https://github.com/dotnet/runtime/issues/81382 -->
+  <ItemGroup>
+    <FrameworkReference Update="Microsoft.NETCore.App"
+                        RuntimeFrameworkVersion="$(MicrosoftNETCoreAppRuntimeVersion)" />
+
+    <PackageReference Include="Microsoft.Dotnet.ILCompiler"
+                      Version="$(MicrosoftNETCoreAppRuntimeVersion)" />
+  </ItemGroup>
+
+  <!--
+  Reference the Microsoft.AspNetCore.App shared framework assemblies with ProjectReference.
+  This allows for easy dev workflow. Just need to update the product code, and re-run the tests.
+  -->
+  <ItemGroup>
+    <ProjectReference Include="$(RepoRoot)src\DefaultBuilder\src\Microsoft.AspNetCore.csproj" />
```

Comment:
Don't we also need to remove the existing FrameworkReference for this to work? Or does conflict resolution automatically picks the greater version here?

Reply:
So I waffled back and forth on how I wanted to do this. High-level there are 2 options:

1. Reference the ASP.NET assemblies through the "shared framework" that is produced by the full build.
    - This is how customers make ASP.NET apps.
    - However this makes the dev workflow less than ideal. ASP.NET doesn't have the "bin place" logic that dotnet/runtime has. So in order to change code and re-run the tests, you would need to do a full build, or copy files to the right spot.
2. Don't be a "Web" SDK project, just a plain Microsoft.NET.Sdk and ProjectReference each ASP.NET assembly
    - This is how our existing tests and samples work

I choose to go with (2) because of the simplicity and because of the better dev workflow.

Reply:
Makes sense. For pure education purposes, do you know what else is different apart from reference assemblies when not being a web sdk project? For instance, does it change how dotnet run works?

Reply:
One example of things that are different is that some MSBuild properties get defaulted differently. For example "Server GC" gets enabled. Currently the "disable Reflection in JsonSerializer" feature switch gets defaulted in the Web SDK as well.

For these cases, I'm not too concerned. For the feature switches, we can default them as needed in the tests, or in the infrastructure.